src/components/Dropdown/Dropdown.tsx

```diff
@@ -162,20 +172,33 @@ function useFocusContextInstance(
   lastTypedOrSubmittedValue: string,
   setValue: (newValue: string) => void,
   screenReaderKey: number,
-  setScreenReaderKey: (newKey: number) => void
+  setScreenReaderKey: (newKey: number) => void,
+  alwaysSelectOption: boolean
 ): FocusContextType {
   const [focusedIndex, setFocusedIndex] = useState(-1);
   const [focusedValue, setFocusedValue] = useState<string | null>(null);
   const [focusedItemData, setFocusedItemData] = useState<Record<string, unknown> | undefined>(undefined);
 
+  useEffect(() => {
+    if (alwaysSelectOption && items.length > 0) {
+      setFocusedIndex(0);
+      setFocusedItemData(items[0].itemData);
+    }
+  }, [alwaysSelectOption, items]);
+
   function updateFocusedItem(updatedFocusedIndex: number, value?: string) {
     const numItems = items.length;
     let updatedValue;
     if (updatedFocusedIndex === -1 || updatedFocusedIndex >= numItems || numItems === 0) {
       updatedValue = value ?? lastTypedOrSubmittedValue;
-      setFocusedIndex(-1);
-      setFocusedItemData(undefined);
-      setScreenReaderKey(screenReaderKey + 1);
+      if (alwaysSelectOption && numItems !== 0) {
+        setFocusedIndex(0);
+        setFocusedItemData(items[0].itemData);
```

Comment:
I think the screenreader key needs to be set here as well as in the `else`. otherwise, the screenreader may not make announcements if the user types but the number of dropdown options stays the same